integrations/oci/sdk/runtime/src/main/java/io/helidon/integrations/oci/sdk/runtime/OciAuthenticationDetailsProvider.java

```diff
@@ -51,23 +51,27 @@
  * This (overridable) provider will provide the default implementation for {@link AbstractAuthenticationDetailsProvider}.
  *
  * @see OciExtension
+ * @see OciConfigBlueprint
  * @see OciConfig
  */
 @Singleton
 @Weight(ServiceInfoBasics.DEFAULT_INJECT_WEIGHT)
 class OciAuthenticationDetailsProvider implements InjectionPointProvider<AbstractAuthenticationDetailsProvider> {
     static final System.Logger LOGGER = System.getLogger(OciAuthenticationDetailsProvider.class.getName());
 
-    static final String TAG_AUTO = "auto";
-    static final String TAG_CONFIG = "config";
-    static final String TAG_CONFIG_FILE = "config-file";
-    static final String TAG_INSTANCE_PRINCIPALS = "instance-principals";
-    static final String TAG_RESOURCE_PRINCIPALS = "resource-principals";
+    static final String KEY_AUTH_STRATEGY = "auth-strategy";
+    static final String TAG_RESOURCE_PRINCIPAL_VERSION = "OCI_RESOURCE_PRINCIPAL_VERSION";
+    static final String VAL_AUTO = "auto";
+    static final String VAL_CONFIG = "config";
+    static final String VAL_CONFIG_FILE = "config-file";
+    static final String VAL_INSTANCE_PRINCIPALS = "instance-principals";
+    static final String VAL_RESOURCE_PRINCIPALS = "resource-principals";
 
-    static final List<String> ALL_STRATEGIES = List.of(TAG_CONFIG,
-                                                       TAG_CONFIG_FILE,
-                                                       TAG_INSTANCE_PRINCIPALS,
-                                                       TAG_RESOURCE_PRINCIPALS);
+    // order is important here - see the tests and the docs
+    static final List<String> ALL_STRATEGIES = List.of(VAL_INSTANCE_PRINCIPALS,
+                                                       VAL_RESOURCE_PRINCIPALS,
+                                                       VAL_CONFIG,
+                                                       VAL_CONFIG_FILE);
```

Comment:
If you explicitly set `auth-strategy: "config"` then it will be used. The availability checks should only apply when scanning (i.e., undefined defaulting to `auto`) or when more than 1 value is configured.

Reply:
One needs to set `auth-strategy` plain and simple.  Think of it like `enabled=true`.  If that is not set then the sub-configuration is ignored.   No magic.

Reply:
I guess we agree to disagree; feel free to resolve this if you need to

Reply:
Ok.  Does the rest of this api, more specifically `OciExtension.ociConfig()` meet your requirements for the Vault integration?

Reply:
Maybe? I'm not really sure? I don't think I need to use `OciConfig`, right? That is something your stuff needs to use? Or maybe I'm misreading.

At the end of the day my "requirements" are simply that I need a `Supplier` of an `AbstractAuthenticationDetailsProvider` (actually, it turns out, a `BasicAuthenticationDetailsProvider` implementation, since OCI [quietly requires it under the covers](https://github.com/oracle/oci-java-sdk/blob/v3.23.0/bmc-common/src/main/java/com/oracle/bmc/http/signing/internal/DefaultRequestSignerFactory.java#L42-L45)).

If there's an officially blessed recipe that uses this stuff, and it's available this early in the whole config setup dance (i.e at meta-configuration time, when, in my case, MicroProfile Config `ConfigSource`s are being set up), and this approach is where we're headed, I'll use it in #4238. A quick sample one-liner of the recipe to get a `Supplier<? extends BasicAuthenticationDetailsProvider>` using this stuff would be appreciated, but is not necessary.

Reply:
Ok, yes, that is a small function to add to my PR.  Pushed it in the last commit https://github.com/helidon-io/helidon/pull/7352/commits/c416d40a2b27f00958b5523755a2c27fb5c71a38. LMK if that is sufficient.

Reply:
It looks like that will let me put together a `Supplier<? extends BasicAuthenticationDetailsProvider>` so I should be able to work with that.

Reply:
I just pushed this also in `OciExtension`.  Please have a look at my last update.

Reply:
OK. It looks like that will let me put together a Supplier<? extends BasicAuthenticationDetailsProvider> so I should be able to work with that.